relayer/storage/redis-storage.ts

```diff
@@ -61,10 +61,16 @@ export interface RedisConnectionOpts {
   namespace?: string;
 }
 
+export interface ExponentialBackoffOpts {
+  attemptsThreshold: number; // after this many attempts, the delay will be constant
+  delayMs: number;
```

Comment:
I'd change this interface. Because you can't set a max delay using this. If you want to retry 20 times, but never wait more than 1 hour, you can't unless you set the delayMs to something you probably don't want.

In the exponential backoff I'd use a `Math.min`

Reply:
So I'd make it something like: `{initialDelayMs: number; maxDelay: number;}`

Reply:
Also, we could use this PR to enable anybody to pass in a retry function, and we provide one if they don't.

Reply:
@gabzim I addressed all of the suggestions, except for the last one, there I have a question: when you refer to the retry function, is it the exp. backoff function? o am I missing something?

Reply:
I think that what gabi means is that the retry doesn't necessarily need to be exponential. You can receive an strategy that allows the user to decide what the next wait-time should be, or if it shouldn't be retried anymore, and the user can decide using that function wether they want to grow the wait time (might depend on what the error is, for example)
I'm in favor of not complicating the interfaces unless we are need it or someone asks for it (so, for the moment just having a default exponential back-off strategy might do)

Reply:
Which I now see that is what you actually did :)

Reply:
Done, in here: https://github.com/wormhole-foundation/relayer-engine/pull/183